VM/src/lstrlib.cpp

```diff
@@ -1032,6 +1032,11 @@ static int str_format(lua_State* L)
                     break;
                 }
             }
+            case '*':
+            {
+                sprintf(buff, "%s", luaL_tolstring(L, arg, NULL));
```

Comment:
This is invalid, because buff is fixed size. You need to append the string directly to the resulting luaL_buffer instead. Also it looks like you need tests for longer strings (which this wouldn't handle without a crash in CI) and a test for internal nul characters (which this code doesn't handle either as it throws away the length).

Reply:
Also handling of '*' needs a separate flag.

Reply:
> Also handling of '*' needs a separate flag.

What's the best way to create an obvious error if interpolated strings are enabled, but not `%*`?

Reply:
I think the regular error handling should be fine here, string.format will fail at runtime if '*' isn't handled. We can also emit a special error message if the flag is false - but the point is we shouldn't couple these in the code, so that we can first enable '*' everywhere, and then separate enable format strings.

Reply:
The code is now:

```cpp
                size_t length;
                const char* string = luaL_tolstring(L, arg, &length);

                luaL_reservebuffer(&b, length, -1);
                luaL_addlstring(&b, string, length);

                continue; /* skip the `addsize' at the end */
```